File: api/predict.js
export default function handler(req, res) {
    // Enable CORS
    res.setHeader('Access-Control-Allow-Credentials', true);
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET,OPTIONS,PATCH,DELETE,POST,PUT');
    res.setHeader('Access-Control-Allow-Headers', 'X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version');
    
    // Handle OPTIONS method for CORS preflight
    if (req.method === 'OPTIONS') {
      res.status(200).end();
      return;
    }
    
    // Define severity levels
    const classLabels = ['minor', 'moderate', 'severe'];
    const severityMultiplier = {
      'minor': 0.5,
      'moderate': 1.0,
      'severe': 1.5
    };
    
    // Mock price data
    const priceData = {
      "TOYOTA": {
        "Camry": {"door": 12000, "bumper": 8000, "fender": 5000}
      },
      "HONDA": {
        "Civic": {"door": 11000, "bumper": 7500, "fender": 4800}
      },
      "NISSAN": {
        "Rogue": {"door": 10500, "bumper": 7000, "fender": 4500}
      }
    };
    
    function getRecommendations(severity) {
      if (severity === 'minor') {
        return "Can be addressed during routine maintenance";
      } else if (severity === 'moderate') {
        return "Should be repaired within the next 500 miles";
      } else {
        return "Requires immediate attention for safety reasons";
      }
    }
    
    // For POST requests
    if (req.method === 'POST') {
      try {
        // Generate random severity
        const severity = classLabels[Math.floor(Math.random() * classLabels.length)];
        
        // Extract form data if available
        const body = req.body || {};
        const make = (body.make || 'HONDA').toUpperCase();
        const modelName = body.model || 'Civic';
        const part = (body.defectArea || 'door').toLowerCase();
        
        // Calculate price estimate
        let priceEstimate = "₹5000-₹8000";  // Default
        try {
          if (priceData[make] && priceData[make][modelName]) {
            if (priceData[make][modelName][part]) {
              const basePrice = priceData[make][modelName][part];
              const multiplier = severityMultiplier[severity] || 1.0;
              const estimatedCost = Math.floor(basePrice * multiplier);
              priceEstimate = `₹${estimatedCost}`;
            }
          }
        } catch (error) {
          console.error("Price calculation error:", error);
        }
        
        // Return response
        res.status(200).json({
          severity: severity.toUpperCase(),
          confidence: (Math.random() * 0.2 + 0.75).toFixed(2),
          priceEstimate: priceEstimate,
          recommendations: getRecommendations(severity)
        });
        
      } catch (error) {
        console.error("API error:", error);
        res.status(500).json({ error: 'Something went wrong' });
      }
    } else {
      // For non-POST methods
      res.status(405).json({ error: 'Method not allowed' });
    }
  }

File: api/predict.py
from http.server import BaseHTTPRequestHandler
import json
import random

# Define severity levels
class_labels = ['minor', 'moderate', 'severe']
severity_multiplier = {
    'minor': 0.5,
    'moderate': 1.0,
    'severe': 1.5
}

# Mock price data
price_data = {
    "TOYOTA": {
        "Camry": {"door": 12000, "bumper": 8000, "fender": 5000, "light": 3000, "windshield": 15000, "bonnet": 10000, "dickey": 9000}
    },
    "HONDA": {
        "Civic": {"door": 11000, "bumper": 7500, "fender": 4800, "light": 2800, "windshield": 14000, "bonnet": 9500, "dickey": 8500}
    },
    "NISSAN": {
        "Rogue": {"door": 10500, "bumper": 7000, "fender": 4500, "light": 2500, "windshield": 13500, "bonnet": 9000, "dickey": 8000}
    }
}

def get_recommendations(severity):
    if severity == 'minor':
        return "Can be addressed during routine maintenance"
    elif severity == 'moderate':
        return "Should be repaired within the next 500 miles"
    else:
        return "Requires immediate attention for safety reasons"

def handler(request):
    # Handle OPTIONS request for CORS
    if request.method == 'OPTIONS':
        return {
            'status': 200,
            'headers': {
                'Access-Control-Allow-Origin': '*',
                'Access-Control-Allow-Headers': 'Content-Type',
                'Access-Control-Allow-Methods': 'POST, OPTIONS'
            }
        }
    
    # For POST requests
    if request.method == 'POST':
        # Generate random severity
        severity = random.choice(class_labels)
        
        # Get form data if available
        try:
            body = json.loads(request.body)
            make = body.get('make', 'HONDA').upper()
            model_name = body.get('model', 'Civic')
            part = body.get('defectArea', 'door').lower()
        except:
            make = 'HONDA'
            model_name = 'Civic'
            part = 'door'
        
        # Calculate price estimate
        price_estimate = "₹5000-₹8000"  # Default
        try:
            if make in price_data and model_name in price_data[make]:
                if part in price_data[make][model_name]:
                    base_price = price_data[make][model_name][part]
                    multiplier = severity_multiplier.get(severity, 1.0)
                    estimated_cost = int(base_price * multiplier)
                    price_estimate = f"₹{estimated_cost}"
        except:
            pass
        
        return {
            'status': 200,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({
                'severity': severity.upper(),
                'confidence': round(random.uniform(0.75, 0.95), 2),
                'priceEstimate': price_estimate,
                'recommendations': get_recommendations(severity)
            })
        }
    
    # For other methods (GET, etc.)
    return {
        'status': 405,
        'headers': {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Origin': '*'
        },
        'body': json.dumps({'error': 'Method not allowed'})
    }